python/lsst/ip/isr/isrStatistics.py

```diff
@@ -0,0 +1,212 @@
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import numpy as np
+import lsst.afw.math as afwMath
+import lsst.afw.image as afwImage
+import lsst.pipe.base as pipeBase
+import lsst.pex.config as pexConfig
+
+from lsst.afw.cameraGeom import ReadoutCorner
+
+
+class IsrStatisticsTaskConfig(pexConfig.Config):
+    """Overscan correction options.
+    """
+    doCtiStatistics = pexConfig.Field(
+        dtype=bool,
+        doc="Measure CTI statistics from image and overscans?",
+        default=False,
+    )
+    stat = pexConfig.Field(
+        dtype=str,
+        default='MEANCLIP',
+        doc="Statistic name to use to measure regions.",
+    )
+    clip = pexConfig.Field(
+        dtype=float,
+        default=3.0,
+        doc="Clipping threshold for background",
+    )
+    nIter = pexConfig.Field(
+        dtype=int,
+        default=3,
+        doc="Clipping iterations for background",
+    )
+    badMask = pexConfig.ListField(
+        dtype=str,
+        default=["BAD", "INTRP", "SAT"],
+        doc="Mask planes to ignore when identifying source pixels."
+    )
+
+
+class IsrStatisticsTask(pipeBase.Task):
+    """Task to measure arbitrary statistics on ISR processed exposures.
+
+    The goal is to wrap a number of optional measurements that are
+    useful for calibration production and detector stability.
+    """
+    ConfigClass = IsrStatisticsTaskConfig
+    _DefaultName = "isrStats"
+
+    def __init__(self, statControl=None, **kwargs):
+        super().__init__(**kwargs)
+        self.allowDebug = True
+
+        self.statControl = afwMath.StatisticsControl(self.config.clip, self.config.nIter,
+                                                     afwImage.Mask.getPlaneBitMask(self.config.badMask))
+        self.statType = afwMath.stringToStatisticsProperty(self.config.stat)
+
+    def run(self, inputExp, ptc=None, overscanResults=None, **kwargs):
+        """Task to run arbitrary statistics.
+
+        The statistics should be measured by individual methods, and
+        add to the dictionary in the return struct.
+
+        Parameters
+        ----------
+        inputExp : `~lsst.afw.image.Exposure`
```

Comment:
I believe it shortens the type in some locations (`Exposure` instead of `lsst.afw.image.Exposure`).  There are uses in the dev guide, but it's not explained as far as I can find.  I've removed them here.

Reply:
Yes, @czwa is correct, although we shouldn't use `~` for the parameter classes since we are generally always going to want the full name there. They are very useful in sphinx docs though so you can write the full class for linking purposes but have it display only the short form in the text.

Reply:
Nice, thanks.